Source/devices/oni/rhythm-api/rhd2000ONIboard.cpp

```diff
@@ -1204,43 +1206,80 @@ bool Rhd2000ONIBoard::isI2cCapable(oni_dev_idx_t device)
     return val > 0;
 }
 
-int Rhd2000ONIBoard::readByte(uint32_t address, oni_reg_addr_t i2cRawAddress, oni_reg_val_t* value, bool sixteenBitAddress)
+int Rhd2000ONIBoard::readByte (oni_dev_idx_t device, uint32_t address, oni_reg_val_t* value)
+{
+    if (! ctx)
+        return 0;
+
+    int result = oni_read_reg (ctx, device, address, value);
+
+    return result;
+}
+
+int Rhd2000ONIBoard::readEepromByte (oni_dev_idx_t i2cRawAddress, uint32_t address, uint8_t& byte)
 {
-    uint32_t i2cAddress = 0x50; 
+    uint32_t i2cAddress = 0x50;
 
     uint32_t registerAddress = (address << 7) | (i2cAddress & 0x7F);
-    registerAddress |= sixteenBitAddress ? 0x80000000 : 0;
+    registerAddress |= 0x80000000;
 
-    int result = oni_read_reg (ctx, i2cRawAddress, registerAddress, value);
+    oni_reg_val_t value;
+
+    int result = readByte (i2cRawAddress, registerAddress, &value);
+
+    if (result == 0)
+        byte = (uint8_t) (value & 0xFF);
+    else
+        byte = 0;
 
     return result;
 }
 
-uint32_t Rhd2000ONIBoard::getDeviceIdOnEeprom (oni_reg_addr_t i2cRawAddress)
+uint32_t Rhd2000ONIBoard::getDeviceIdOnEeprom (const uint32_t port)
 {
-    if (i2cRawAddress != DEVICE_I2C_RAW_A && i2cRawAddress != DEVICE_I2C_RAW_B && i2cRawAddress != DEVICE_I2C_RAW_C && i2cRawAddress != DEVICE_I2C_RAW_D)
+    if (!ctx || port > 3)
         return 0;
 
+    oni_dev_idx_t i2cRawAddress = DEVICE_I2C_RAW_A + port * 2;
+
+    // NB: Confirm EEPROM first four bytes contains OESH
+    const char identifier[] = "OESH";
```

Comment:
just to be sure, make it "OESH\x01" to include the spec version, in case we change it later, so even an outdated plugin would not break.